petition_service/internal/controller/rpc/petition_controller.go

```diff
@@ -228,3 +232,73 @@ func (s *Server) GetUserVotedPetitions(_ context.Context, req *pb.GetUserVotedPe
 		Petitions: getUserPetitionsResponse,
 	}, nil
 }
+
+func (s *Server) CheckIfPetitionsExpired(_ context.Context, req *pb.Petition) (*empty.Empty, error) {
+	petition, err := s.PetitionService.GetByID(uint(req.Id))
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			slog.Errorf("Petition %v not found", req.Id)
+			return nil, status.Error(codes.NotFound, "petition not found")
+		}
+		return nil, err
+	}
+	slog.Info("Petition %v successfully found", req.Id)
+
+	if _, err := s.PetitionService.CheckPetitionExpiration(petition); err != nil {
+		return nil, err
+	}
+
+	slog.Info("Petition %v expiration date successfully checked", req.Id)
+	return &empty.Empty{}, nil
+}
+
+func ScheduleDailyCheck(s *Server) {
+	c := cron.New()
+	slog.Info("Scheduled Expiration Checker successfully started...")
+
+	petitions, err := s.PetitionService.GetAllActive()
```

Comment:
I think it'd be better if instead of getting all petitions at once, it would be better to control the batching here, cause potentially reading mlns of rows into memory doesn't sound too good. Something like 

```
offset = 0
limit = 1000
for {
    batch = getAllWithStatus(status, limit, offset)
    if len(batch) == 0 { break }
   
   //
   go func() {
       for _, pet = range petitions {// add this for loop for checking petitions in separate goroutine}
   }()

   offset += limit
}
```